climada/util/coordinates.py

```diff
@@ -1057,6 +1057,79 @@ def assign_coordinates(coords, coords_to_assign, distance="euclidean",
                 coords_to_assign, coords[not_assigned_idx_mask], threshold, **kwargs)
     return assigned_idx
 
+def assign_haz_centroids(coord_df, hazard, crs, distance='euclidean',
+                        threshold=NEAREST_NEIGHBOR_THRESHOLD,
+                        overwrite=True):
+    """Assign to each exposure's coordinate point its closest hazard's coordinate.
+    If disatances > threshold in points' distances, -1 is returned.
+    If raster hazard and coordinate point outside raster, -1 is returned.
+
+    Parameters
+    ----------
+    coord_df : pd.DataFrame
+        Dataframe with defined latitude/longitude column
+    hazard : Hazard
+        Hazard to match (with raster or vector centroids).
```

Comment:
This method that shouldn't care about CLIMADA data structures unnecessarily. 
I suggest to use `centroids` as an argument instead.

The only other attribute needed is `tag.haz_type` and that's used "in the wrong way" to check whether the centroids have been assigned before. Which is misleading as it's thinkable to have two different hazards with the same type. When returning output data instead of overwriting input data, `tag.haz_type` will be obsolete anyways. (see comment to `overwrite`)

Reply:
Is changed to centroids now